workflow-testing/src/main/java/com/squareup/workflow1/testing/RenderTester.kt

```diff
@@ -19,14 +22,43 @@ import kotlin.reflect.KTypeProjection
  *
  * See [RenderTester] for usage documentation.
  */
+@OptIn(WorkflowExperimentalApi::class) // Opt-in is only for the argument check.
 @Suppress("UNCHECKED_CAST")
 public fun <PropsT, OutputT, RenderingT> Workflow<PropsT, OutputT, RenderingT>.testRender(
   props: PropsT
 ): RenderTester<PropsT, *, OutputT, RenderingT> {
   val statefulWorkflow = asStatefulWorkflow() as StatefulWorkflow<PropsT, Any?, OutputT, RenderingT>
   return statefulWorkflow.testRender(
     props = props,
-    initialState = statefulWorkflow.initialState(props, null)
+    initialState = run {
+      require(this !is SessionWorkflow<PropsT, *, OutputT, RenderingT>) {
+        "Called testRender on a SessionWorkflow without a CoroutineScope. Use the version that passes a CoroutineScope."
+      }
+      statefulWorkflow.initialState(props, null)
+    }
+  ) as RenderTester<PropsT, Nothing, OutputT, RenderingT>
+}
+
+/**
+ * Create a [RenderTester] to unit test an individual render pass of this [SessionWorkflow],
+ * using the workflow's [initial state][StatefulWorkflow.initialState], in the [workflowScope].
+ *
+ * See [RenderTester] for usage documentation.
+ */
+@OptIn(WorkflowExperimentalApi::class)
+@Suppress("UNCHECKED_CAST")
+public fun <PropsT, OutputT, RenderingT> Workflow<PropsT, OutputT, RenderingT>.testRender(
```

Comment:
Can't you make this version an extension on `SessionWorkflow` and avoid the runtime check?

Reply:
ya I thought about that but then I thought about how in your tests you might not have the reified `SessionWorkflow` type? Maybe you were passing it around or dealign with it just as a `Workflow`? Maybe I was overthinking it though. Perhaps we could just make it an extension on `SessionWorkflow` 🤔

Reply:
I think so. I'm going to write a unit test against `class RealFooWorkflow`, not `interface FooWorkflow`.

Reply:
ya, I think you are right. I will change it.

- [ ] testRender + CoroutineScope extension extends `SessionWorkflow` not `Workflow`